auto3dseg/algorithm_templates/segresnet/scripts/dummy_runner.py

```diff
@@ -0,0 +1,180 @@
+from monai.losses import DeepSupervisionLoss
+from torch.cuda.amp import GradScaler, autocast
+
+
+class DummyRunnerSegResNet(object):
+    def __init__(self, output_path, data_stats_file, device_id: int = 0):
+        config_file = []
+        config_file.append(
+            os.path.join(output_path, "configs", "hyper_parameters.yaml")
+        )
+
+        parser = ConfigParser()
+        parser.read_config(config_file)
+
+        self.device = torch.device("cuda:{0:d}".format(device_id))
+        torch.cuda.set_device(self.device)
+
+        self.amp = parser.get_parsed_content("amp")
+        self.input_channels = parser.get_parsed_content("input_channels")
+        self.roi_size = parser.get_parsed_content("roi_size")
+        self.overlap_ratio = 0.625
+        self.num_sw_batch_size = 1
+
+        output_classes = parser.get_parsed_content("output_classes")
+        sigmoid = parser.get_parsed_content("sigmoid")
+        self.label_channels = self.output_classes if sigmoid else 1
+
+        print("roi_size", self.roi_size)
+        print("label_channels", self.label_channels)
+
+        self.model = parser.get_parsed_content("network")
+        self.model = self.model.to(self.device)
+
+        self.loss_function = parser.get_parsed_content("loss")
+        self.loss_function = DeepSupervisionLoss(self.loss_function)
+
+        optimizer_part = parser.get_parsed_content("optimizer", instantiate=False)
+        self.optimizer = optimizer_part.instantiate(params=self.model.parameters())
+
+        with open(data_stats_file) as f_data_stat:
+            data_stat = yaml.full_load(f_data_stat)
+
+        self.max_shape = [0, 0, 0]
+        if parser.get_parsed_content("resample"):
+            pixdim = parser.get_parsed_content("resample_resolution")
+
+            for _k in range(len(data_stat["stats_by_cases"])):
+                image_shape = data_stat["stats_by_cases"][_k]["image_stats"]["shape"]
+                image_shape = np.squeeze(image_shape)
+                image_spacing = data_stat["stats_by_cases"][_k]["image_stats"][
+                    "spacing"
+                ]
+                image_spacing = np.squeeze(image_spacing)
+                image_spacing = [np.abs(image_spacing[_i]) for _i in range(3)]
+
+                new_shape = [
+                    int(
+                        np.ceil(float(image_shape[_l]) * image_spacing[_l] / pixdim[_l])
+                    )
+                    for _l in range(3)
+                ]
+                if np.prod(new_shape) > np.prod(self.max_shape):
+                    self.max_shape = new_shape
+        else:
+            for _k in range(len(data_stat["stats_by_cases"])):
+                image_shape = data_stat["stats_by_cases"][_k]["image_stats"]["shape"]
+                image_shape = np.squeeze(image_shape)
+
+                if np.prod(image_shape) > np.prod(self.max_shape):
+                    self.max_shape = image_shape
+        print("max_shape", self.max_shape)
```

Comment:
you can compute max_shape first, and then account for pix_dim later (if resample)

also this can be inaccurate since we pre-crop images on CPU first and the shape changes. We crop foreground always, and crop a special region for a recent tutorial.  You could use "image_size" from a config (which is an approximate estimate)